jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/TransactionListWithProof.java

```diff
@@ -0,0 +1,28 @@
+package dev.jlibra.admissioncontrol.query;
+
+import org.immutables.value.Value;
+import types.Events;
+
+import java.util.Collection;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Value.Immutable
+public interface TransactionListWithProof {
+    List<Event> getEvents();
+
+    static TransactionWithProof fromGrpcObject(types.TransactionOuterClass.TransactionListWithProof grpcObject) {
+        List<List<Events.Event>> collect = grpcObject.getEventsForVersions().getEventsForVersionList().stream()
+                .map(Events.EventsList::getEventsList)
+                .collect(toList());
+
+        List<Event> eventResult = collect.stream().map(eventGrpcObjects ->
+             eventGrpcObjects.stream().map(Event::fromGrpcObject).collect(toList())
+        ).flatMap(Collection::stream).collect(toList());
```

Comment:
Also here it is unnecessary to `collect(toList())` only to immediately `Collection::stream` in the flatMap. Both can be dropped.

Reply:
Adjusted accordingly